### PR TITLE
Update error strings to fix failing system tests

### PR DESCRIPTION
**Pull request by Wedgetable**
### What does this Pull Request accomplish?

Removes IVI and hex code specification from error strings on 3 tests, and checks for the substring instead:

* `NiFgenSessionTest.InitWithErrorFromDriver_ReturnsUserErrorMessage`
* `NiRFSADriverApiTests.ErrorMessage_ReturnsErrorMessage`
* `NiRFSGDriverApiTests.ErrorMessage_ReturnsErrorMessage`

### Why should this Pull Request be merged?

Fixes 3 system tests that are consistently failing.

### What testing has been done?

Ran tests locally on a machine with drivers - the tests now pass.

### CODE REVIEW

**Comment by DavidCurtiss** [comment]
When I automated the system tests, I found that at least some of these IVI errors will report the hex code on one platform but not the other. If you hit the same thing, then you can remove the "IVI:" prefix, and change the check from EXPECT_EQ to `EXPECT_THAT(..., HAS_SUBSTR(...))`

If you do this, then you'll also need to `#include <gmock/gmock.h>` if it's not already included, and add `using namespace ::testing;`

For reference, you can see my changes to the dcpower tests [here](https://github.com/ni/grpc-device/pull/675/files#diff-311bfa03f686d483ffc820012e65425a6dc1f6e2abc4ae0c785b1f4165171ba8).

**Review by reckenro — APPROVED** [review]

File: source/tests/system/nifgen_session_tests.cpp
Comment (dmondrik) [inline source/tests/system/nifgen_session_tests.cpp:155]: Note that we're both changing code in this area. Second to commit gets to merge it.

File: source/tests/system/nirfsa_driver_api_tests.cpp
Comment (astarche) [inline source/tests/system/nirfsa_driver_api_tests.cpp:604]: This still has the `IVI: ` prefix, so it's not a `substr` of the original message. Shouldn't it be the same as the one below?

**Review by astarche — APPROVED** [review]

File: source/tests/system/nifgen_session_tests.cpp
Comment (Wedgetable) [reply source/tests/system/nifgen_session_tests.cpp:155]: I'm going to wait for this last update to go through CI checks, but I'll PM you before I merge just in case you're about to submit as well.

**Pull request merged**
